### PR TITLE
Add macOS workflow to tests

### PR DESCRIPTION
**Pull request by petricm**
BEGINRELEASENOTES
- add a macOS test to GitHub actions

ENDRELEASENOTES
But for a proper implementation I need https://github.com/actions/virtual-environments/issues/598 now it downloads root from the root webpage

### CODE REVIEW

**Comment by andresailer** [comment]
missing `yaml` python package

**Comment by andresailer** [comment]
Also this clearly points that a check for the return value of podio_class_generator should be added to abort if this fails.

File: .github/workflows/mac.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+jobs:
+  test:
+    runs-on: macos-10.15
+    steps:
+    - uses: actions/checkout@v2
+    - name: Download ROOT
+      run: |
+        curl -O https://root.cern/download/root_v6.20.02.macosx64-10.15-clang110.tar.gz
+        tar xf root_v6.20.02.macosx64-10.15-clang110.tar.gz
+        sudo pip install yaml
```
Comment (andresailer) [inline .github/workflows/mac.yml]:
```suggestion
        sudo pip install pyyaml
```

Comment (petricm) [reply .github/workflows/mac.yml]: I am checking what else is missing, this needs a requirements.txt file

Comment (petricm) [reply .github/workflows/mac.yml]: ok all imports 
```py
set(['string', 're', 'optparse', 'ROOT', 'unittest', 'subprocess', 'yaml', 'podio_config_reader', 'collections', 'EventStore', 'pickle', 'podio_templates', 'copy', 'os'])
```
only `yaml` is not in the basic python or ROOT or podio

**Pull request merged**